python/chainer_compiler.py

```diff
@@ -159,18 +178,28 @@ def __init__(self, model, inputs, dump_onnx=False):
             self.compile(inputs)
 
     def compile(self, inputs):
-        xmodel = ch2o.compile_model(self.mc, inputs)
-        f = tempfile.NamedTemporaryFile(delete=False)
-        f.write(xmodel.SerializeToString())
-        f.close()
-        del xmodel
+        if self.translator == 'ch2o':
+            xmodel = ch2o.compile_model(self.mc, inputs)
+            f = tempfile.NamedTemporaryFile(delete=False)
+            f.write(xmodel.SerializeToString())
+            f.close()
+            del xmodel
+        elif self.translator == 'onnx_chainer':
+            import onnx_chainer
+            f = tempfile.NamedTemporaryFile(delete=False)
+            onnx_chainer.export(self.mc, inputs, filename=f)
+            f.close()
+        else:
+            raise NotImplementedError('Unsupported translator:',
+                                      self.translator)
 
         graph = chainer_compiler_core.load(f.name)
         os.unlink(f.name)
 
         self.orig_output_names = graph.output_names()
 
-        fwd_graph, bwd_graph = graph.backward_to(graph.input_names())
+        # fwd_graph, bwd_graph = graph.backward_to(graph.input_names())
+        fwd_graph, bwd_graph = graph.backward_to(graph.param_names())
```

Comment:
I think gradient nodes should be generated not only for params but also for inputs since chainer_compiler.compile may be called to compile only a part of a whole graph. For example, if a user compiles only ResBlock 3, gradients of inputs should be computed for backprop of ResBlock 1 and 2.